inc/fluent_bit/error_handler/namespace.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * PHP Error Handler to send all PHP errors to CloudWatch
+ *
+ * We do this to get better structured data about the errors, and also tie XRay trace ids to the errors.
+ *
+ * @package altis/cloud
+ */
+
+namespace Altis\Cloud\Fluent_Bit\Error_Handler;
+
+use Altis\Cloud\Fluent_Bit;
+
+/**
+ * Set up shutdown function error handler to send to CloudWatch.
+ *
+ * @return void
+ */
+function bootstrap() {
+	$GLOBALS['altis_cloudwatch_error_handler_errors'] = [];
+	$GLOBALS['altis_cloudwatch_error_handler_error_count'] = 0;
+
+	// If there is already an error handler set, we want to make sure it's not lost.
+	$current_errror_handler = set_error_handler( function () use ( &$current_errror_handler ) { // @codingStandardsIgnoreLine
+		call_user_func_array( __NAMESPACE__ . '\\error_handler', func_get_args() );
+		if ( $current_errror_handler ) {
+			return call_user_func_array( $current_errror_handler, func_get_args() );
+		}
+		return false;
+	} );
+
+	// Hook into Query Monitor error handler in case the above is overridden.
+	add_action( 'qm/collect/new_php_error', __NAMESPACE__ . '\\error_handler', 10, 5 );
+}
+
+/**
+ * Shutdown function error handler callback.
+ *
+ * @param integer $errno Error code.
+ * @param string $errstr Error message.
+ * @param string $errfile The file where the error occurred.
+ * @param integer $errline The line on which the error occurred.
+ * @return boolean
+ */
+function error_handler( int $errno, string $errstr, string $errfile = null, int $errline = null ) : bool {
+	global $altis_cloudwatch_error_handler_errors, $altis_cloudwatch_error_handler_error_count;
+	// Limit the amount of errors to hold in memory. Flush every 100.
+
+	$error = [
+		'type'    => get_error_type_for_error_number( $errno ),
+		'message' => $errstr,
+		'file'    => str_replace( dirname( ABSPATH ), '', $errfile ),
+		'line'    => $errline,
+	];
+	$error = apply_filters( 'altis_cloudwatch_error_handler_error', $error );
+	$json = json_encode( $error );
+
+	$logger = Fluent_Bit\get_logger( 'app.php-structured.' . $error['type'] );
+	$logger->error( $json );
+
+	// TODO is this necessary to keep?
+	$altis_cloudwatch_error_handler_errors[ $errno ][] = [
+		'timestamp' => time() * 1000,
+		'message'   => $json, // @codingStandardsIgnoreLine
+	];
+	$altis_cloudwatch_error_handler_error_count++;
```

Comment:
With Fluent Bit, we don't need to send logs in batches. Is this necessary anymore?

Reply:
@joehoyle do you know if this isn't needed anymore?